NGitLab/Impl/MilestoneClient.cs

```diff
@@ -9,10 +9,10 @@ public class MilestoneClient : IMilestoneClient
         private readonly API _api;
         private readonly string _milestonePath;
 
-        public MilestoneClient(API api, int projectId)
+        public MilestoneClient(API api, int projectId, string urlPrefix = "projects")
```

Comment:
I don't like the idea of being able to create a `MilestoneClient` with just any `urlPrefix`...

Since a client application should always go through `IGitLabClient.GetMilestone()` or `IGitLabClient.GetGroupMilestone()` to create a `MilestoneClient`, I think we should instead introduce an internal ctor. Something like

```cs
internal MilestoneClient(API api, MilestoneScope scope, int id)
```

where `MilestoneScope` would be an enum that says whether we're interested in group or project milestones. Of course, `id` would then represent a group or project, accordingly.